RevenueCatUI/UIKit/PaywallFooterViewController.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PaywallFooterViewController.swift
+//
+//  Created by Antonio Rico Diez on 4/12/23.
+//
+
+#if canImport(UIKit) && !os(tvOS) && !os(watchOS)
+
+import RevenueCat
+import SwiftUI
+import UIKit
+
+/// A view controller for displaying `PaywallData` for an `Offering`.
+///
+/// - Seealso: ``PaywallView`` for `SwiftUI`.
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, *)
+@objc(RCPaywallFooterViewController)
+public final class PaywallFooterViewController: UIViewController {
```

Comment:
There's a lot of duplication here with `PaywallViewController`. I think we can implement this by subclassing `PaywallViewController` instead and overriding the `mode`.